src/io/overlapped.cr

```diff
@@ -8,19 +8,21 @@ module IO::Overlapped
     property fiber : Fiber?
   end
 
-  def self.wait_queued_completions(timeout, &)
+  def self.wait_queued_completions(timeout, alertable = false, &)
     overlapped_entries = uninitialized LibC::OVERLAPPED_ENTRY[1]
 
     if timeout > UInt64::MAX
       timeout = LibC::INFINITE
     else
       timeout = timeout.to_u64
     end
-    result = LibC.GetQueuedCompletionStatusEx(Crystal::EventLoop.current.iocp, overlapped_entries, overlapped_entries.size, out removed, timeout, false)
+    result = LibC.GetQueuedCompletionStatusEx(Crystal::EventLoop.current.iocp, overlapped_entries, overlapped_entries.size, out removed, timeout, alertable)
     if result == 0
       error = WinError.value
       if timeout && error.wait_timeout?
         return true
+      elsif alertable && error.error_exe_marked_invalid?
```

Comment:
**issue:** `ERROR_EXE_MARKED_INVALID` for queing an APC seems very strange.
I understand this is the observed error value, but is there any confirming documentation?

Reply:
Aaaah, looks like this is actually supposed to be interpreted as `WAIT_IO_COMPLETION`. It has the same value as `ERROR_EXE_MARKED_INVALID` 🤦 

```suggestion
      elsif alertable && error.value == LibC::WAIT_IO_COMPLETION
```
Thanks to this SO comment: https://stackoverflow.com/questions/18691039/getqueuedcompletionstatusex-behavior-when-alerted#comment69398998_18709932

Reply:
Thank you!

Searching for `WAIT_IO_COMPLETION` the waiting functions (that don't include `GetQueuedCompletionStatusEx`) are supposed to RETURN that value, and nowhere do they way that `GetQueuedCompletionStatusEx` will report it as `GetLastError` that... doesn't always return a `WinError`... :face_in_clouds:

Reply:
Yeah, this is super strange. Win32 API docs are usually quite extensive. But there's always a limit somewhere 😆